src/update.rs

```diff
@@ -288,6 +289,29 @@ where
     }
 }
 
+unsafe impl<T> Update for Arc<T>
+where
+    T: Update,
+{
+    unsafe fn maybe_update(old_pointer: *mut Self, new_arc: Self) -> bool {
```

Comment:
I wonder if it would make sense to do a ptr equality check at the start and return `false` if they are equal. That probably is a very unlikely scenario (usually you will compute an entirely new arc allocation I believe?).